internal/cmd/cliopts/flat.go

```diff
@@ -67,6 +67,13 @@ func (w *flatSourceWalker) Exit(loc reflectwalk.Location) error {
 	return nil
 }
 
+func (w *flatSourceWalker) SliceElem(_ int, value reflect.Value) error {
+	if value.Type().Kind() == reflect.Struct || isPtrToStruct(value) {
+		w.location = append(w.location, "")
```

Comment:
This `w.location` is effectively a stack that keeps track of the location within the Go structure that is receiving configuration value. 

Above here in this file there is an `Exit` method that pops the last item off of `w.location`. What was happening was that a slice of structs was causing that pop , but there was no matching "push", so the location was wrong.

This fixes the problem by pushing an empty items onto `location` for each item in the slice. This way the `Exit` will correctly pop the empty item, and the `location` value will be correct after iterating over the slice is complete.

We were already using this empty string value to indicate embedded structs, so this re-uses it for slice elements. Both of these things are treated the same way because they don't modify the path at which we expect a value.